support/vm/vector_gen.go

```diff
@@ -52,6 +53,17 @@ func (g *vectorGen) before(v *VM, name string) error {
 	return nil
 }
 
+func (g *vectorGen) fillCar(v *VM) error {
+	var err error
+	getter := nodeGetterFromStore(v.Store())
+	roots := []cid.Cid{g.vector.StartStateTree}
+	if g.conformance() {
+		roots = append(roots, g.vector.EndStateTree)
+	}
+	g.vector.State, err = encodeCAR(getter, roots...)
```

Comment:
Rather than directly manipulate `vector.State`, it would be more consistent to follow the pattern of adding a `SetState()` option-style method. Other such method already take the VM as a parameter, so this logic could move there.